dbms/src/Functions/FunctionsConversion.h

```diff
@@ -1138,6 +1138,10 @@ struct ToIntMonotonicity
 
     static IFunction::Monotonicity get(const IDataType & type, const Field & left, const Field & right)
     {
+        /// If type is string, the function is no monotonic
+        if (WhichDataType(type).isString())
```

Comment:
1. What about other types: FixedString for example.
Maybe better to specify condition with `isRepresentedByNumber`?

2. Conversion from String to String; from Array to Array of same type is monotonic.

Reply:
> What about other types: FixedString for example.
> Maybe better to specify condition with isRepresentedByNumber?

Done

> Conversion from String to String; from Array to Array of same type is monotonic.

ToIntMonotonicity is just use to `any -> Int`, So I'm not sure if it has `Array -> Array` 

And I rewrite the  test to make it testable.